app/src/main/java/com/x8bit/bitwarden/data/platform/manager/BiometricsEncryptionManagerImpl.kt

```diff
@@ -89,6 +94,8 @@ class BiometricsEncryptionManagerImpl(
 
     /**
      * Generates a [SecretKey] from which the [Cipher] will be generated.
+     *
+     * @throws ProviderException if [KeyGenerator] is unable to generate a key.
```

Comment:
Is there a reason we don't do all the exception handling inside this `generateKey` function itself? We really want to avoid having any functions that we know throw exceptions. This could be, like `generateKeyOrNull()`

cc: @david-livefront

Reply:
Thanks for making me dig into this a little deeper. Originally, I wanted to prevent a cascade of other changes needed to handle a nullable return value from this function. Main in `createCipher`.

Now that I'm taking a second look, I'm seeing that some call sites of `createCipher` may not be handled safely. That's making me think the changes I was trying to avoid are needed regardless. Otherwise `createCipher` is going to cause the same crash. 🤔